### ISSUE TITLE
[DocDB] [xCluster-Atomicity] Data loss at target after rolling restart

### ISSUE THREAD

**Issue by Arjun-yb** [yugabyte/yugabyte-db#15250]
Jira Link: [DB-4421](https://yugabyte.atlassian.net/browse/DB-4421)
### Description

**Steps**:

1. Create 2 universes(Source and Target) with Turn ON the GFlag `enable_replicate_transaction_status_table` on master of both universes
2. Create database and tables at both sides
```
create table table1(id int primary key, description text);
create table table2(id int primary key, description text);
```
3. Setup replication from source to target
4. Change the xCluster role at Target
`yb-admin -master_addresses <master-address> change_xcluster_role STANDBY`
5. Run transactional workload, one transaction with 200K rows - (Begin transaction; insert 100K rows in table1 and 100K rows in table2; Commit;)
6. Validate data at both sides
7. Do rolling restart(both universes)
8. Again validate data at both sides.

**Observations**:
1. After rolling restart data at target side are missing(initially both tables had 100K rows/table) and after rolling restart both tables have < 40% - 70% data.

**Comment by YB-WangTx** [comment]
Hum, I was not be able to reproduce it, loaded 3 tables and 3 indexes, row count match before and after rolling restart. Will check out more...

**Comment by lingamsandeep** [comment]
Fixed by 64a7b3aa96894604689ed7f7a1f74e52078b1c57.
